onnxruntime/test/python/onnxruntime_test_ort_trainer.py

```diff
@@ -406,6 +407,47 @@ def testMNISTSaveAsONNX(self):
         trainer.save_as_onnx(onnx_file_name)
         assert os.path.exists(onnx_file_name)
 
+    def testMNISTInitializerNames(self):
+        torch.manual_seed(1)
+        device = torch.device("cuda")
+
+        mnist = MNISTWrapper()
+        train_loader, test_loader = mnist.get_loaders()
+        model, model_desc = mnist.get_model()
+
+        trainer = mnist.get_trainer(model, model_desc, device)
+        learningRate = 0.02
+        epoch = 0
+
+        data, target = next(iter(train_loader))
+        data, target = data.to(device), target.to(device)
+        data = data.reshape(data.shape[0], -1)
+
+        loss, _ = trainer.train_step(data, target, torch.tensor([learningRate]))
+
+        assert set([n.name for n in trainer.onnx_model_.graph.initializer]) \
+            == set([n for n, t in model.named_parameters()])
+
+    def testMNISTFrozenWeight(self):
+        torch.manual_seed(1)
+        device = torch.device("cuda")
+
+        mnist = MNISTWrapper()
+        train_loader, test_loader = mnist.get_loaders()
+        model, model_desc = mnist.get_model()
+
+        trainer = mnist.get_trainer(model, model_desc, device, frozen_weights=['fc1.weight'])
+
+        learningRate = 0.02
+        epoch = 0
+
+        data, target = next(iter(train_loader))
+        data, target = data.to(device), target.to(device)
+        data = data.reshape(data.shape[0], -1)
+
+        loss, _ = trainer.train_step(data, target, torch.tensor([learningRate]))
+        assert 'fc1.weight' not in trainer.state_dict() and 'fc2.weight' in trainer.state_dict()
```

Comment:
Does that mean that if I pass a 'frozen_weight', it would be excluded from the ORT state dict after training? That means in a scenario where I want to use the ORT trained model, I would also need the previous pre-trained model that I used to initialize the 'frozen_weight' in order to get the complete state. Is it possible to include these frozen_weights in the state_dict after calling model.state_dict()?

Reply:
What you mentioned about the state_dict is exactly correct. I also think state_dict should include all the weights. 
I wasn't sure if it was intentionally designed like this, I'll check with Sherlock or Cheng.

Reply:
@ashbhandare checked with Sherlock, I'll also udpate state_dict in this PR.

Reply:
As a side note, in the new design, if instead of `frozen_weights`, the user creates a parameter group on the OptimizerConfig with all hyper params zeroed, state dict would still contain the non-trainable weights in the output